mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ModelPropertiesDeclarationUtils.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.util;
+
+import org.mule.runtime.api.meta.model.EnrichableModel;
+import org.mule.runtime.api.meta.model.ModelProperty;
+import org.mule.runtime.extension.internal.property.ComposedOperationModelProperty;
+import org.mule.runtime.extension.internal.property.NoConnectivityErrorModelProperty;
+import org.mule.runtime.extension.internal.property.NoErrorMappingModelProperty;
+import org.mule.runtime.extension.internal.property.NoReconnectionStrategyModelProperty;
+import org.mule.runtime.extension.internal.property.NoStreamingConfigurationModelProperty;
+import org.mule.runtime.extension.internal.property.NoTransactionalActionModelProperty;
+import org.mule.runtime.extension.internal.property.PagedOperationModelProperty;
+import org.mule.runtime.extension.internal.property.TargetModelProperty;
+import org.mule.runtime.extension.internal.property.TransactionalActionModelProperty;
+import org.mule.runtime.extension.internal.property.TransactionalTypeModelProperty;
+
+import java.util.Set;
+
+/**
+ * Provides a way to access the functionality on internal model properties when declaring crafted extensions.
+ * 
+ * @since 1.8
+ */
+public final class ModelPropertiesDeclarationUtils {
```

Comment:
this entire class reveals the fact that certain concerns are implemented through model properties, which defeats the purpose. Instead, change the semantics to be centered on the intent. For example: instead of `public static final ModelProperty noStreamingConfigurationModelProperty()` -> `public static void withNoStreamingConfiguration(ComponentDeclarer)`.

This not only affects the method signatures but also the class name. Also, is this fine being a class with static methods or should it be an interface which impl is discovered? I would also avoid the `Utils` suffix and follow a pattern similar to `org.mule.runtime.extension.api.metadata.ComponentMetadataConfigurer`

Reply:
random idea: what if this is not an utils at all and instead these methods are added on the declarers directly? Pros/cons?

Reply:
would be nice, but not doable since the declarer is in mule-api and the model properties ar ein extensions-api